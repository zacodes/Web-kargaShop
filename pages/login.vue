<template>
  <div class="container" id="Home">
    <div class="row">
      <div class="col-4 offset-4 border my-4" style="border-radius: 5px">
        <div class="row">
          <div class="col-12 text-center mt-3">
            <i class="fa fa-user-circle" style="font-size:60px; color:#ddd"></i>
          </div>
          <div class="col-12 text-center mt-3 text-ml">
            Üye Girişi
          </div>
          <div class="col-12 mt-3">
            <b-input-group prepend="@">
              <b-form-input placeholder="Email"></b-form-input>
            </b-input-group>
          </div>
          <div class="col-12 mt-3">
            <b-input-group>
              <b-input-group-prepend>
                <div class="input-group-text" data-children-count="0">
                  <i class="fa fa-lock"></i>
                </div>
              </b-input-group-prepend>
              <b-form-input placeholder="Şifre"></b-form-input>
            </b-input-group>
          </div>
          <div class="col-6 mt-3">
            <div class="form-check form-check-inline">
              <input class="form-check-input" type="checkbox" id="inlineCheckbox1" value="option1">
              <label class="form-check-label" for="inlineCheckbox1">Beni Hatırla</label>
            </div>
          </div>
          <div class="col-6 mt-3 text-right">
            <a href="">Şifremi Unuttum</a>
          </div>
          <div class="col-12 my-3">
            <a href="" class="btn btn-warning btn-block" style="border-radius: 30px; color: #fff">
              Giriş Yap
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>
