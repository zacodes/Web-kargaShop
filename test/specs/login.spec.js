import { mount, shallowMount } from '@vue/test-utils'
import login from "../../pages/login.vue"

describe('login.vue', () => {

    it('login.vue Bileşenimizin class özelliğini kontrol edilmesi', () => {
        const wrapper = mount(login)
        expect(wrapper.classes()).toContain('container');
    });

    it("yazitest", () => {
        const wrapper = mount(login)
        expect(wrapper.text()).toContain("Üye Girişi");
    });

    it("yazitest2", () => {
        const wrapper = mount(login)
        expect(wrapper.text()).toContain("Beni Hatırla");
    });

    it("yazitest3", () => {
        const wrapper = mount(login)
        expect(wrapper.text()).toContain("Şifremi Unuttum");
    });

    it("yazitest4", () => {
        const wrapper = mount(login)
        expect(wrapper.text()).toContain("Giriş Yap");
    });

    it('Snapshot özelliği', () => {
        const wrapper = mount(login)
        expect(wrapper.html()).toMatchSnapshot()
    });

    it('has data', () => {
        const wrapper = mount(login)
        expect(typeof login.data).toBe('function')
    });
})

