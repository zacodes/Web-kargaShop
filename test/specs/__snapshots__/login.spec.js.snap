// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login.vue Snapshot özelliği 1`] = `
<div id="Home" class="container">
  <div class="row">
    <div class="col-4 offset-4 border my-4" style="border-radius: 5px;">
      <div class="row">
        <div class="col-12 text-center mt-3"><i class="fa fa-user-circle" style="font-size: 60px; color: rgb(221, 221, 221);"></i></div>
        <div class="col-12 text-center mt-3 text-ml">
          Üye Girişi
        </div>
        <div class="col-12 mt-3">
          <b-input-group prepend="@">
            <b-form-input placeholder="Email"></b-form-input>
          </b-input-group>
        </div>
        <div class="col-12 mt-3">
          <b-input-group>
            <b-input-group-prepend>
              <div data-children-count="0" class="input-group-text"><i class="fa fa-lock"></i></div>
            </b-input-group-prepend>
            <b-form-input placeholder="Şifre"></b-form-input>
          </b-input-group>
        </div>
        <div class="col-6 mt-3">
          <div class="form-check form-check-inline"><input type="checkbox" id="inlineCheckbox1" value="option1" class="form-check-input"> <label for="inlineCheckbox1" class="form-check-label">Beni Hatırla</label></div>
        </div>
        <div class="col-6 mt-3 text-right"><a href="">Şifremi Unuttum</a></div>
        <div class="col-12 my-3"><a href="" class="btn btn-warning btn-block" style="border-radius: 30px; color: rgb(255, 255, 255);">
            Giriş Yap
          </a></div>
      </div>
    </div>
  </div>
</div>
`;
